src/posit/connect/external/databricks.py

```diff
@@ -395,42 +345,59 @@ def text():
 
     def __init__(
         self,
-        local_strategy: CredentialsStrategy,
+        local_strategy: Optional[CredentialsStrategy] = None,
+        workbench_strategy: Optional[CredentialsStrategy] = None,
+        connect_strategy: Optional[CredentialsStrategy] = None,
         client: Optional[Client] = None,
-        user_session_token: Optional[str] = None,
     ):
+        self._auth_type = "posit-connect-default"
         self._local_strategy = local_strategy
+        self._workbench_strategy = workbench_strategy
+        self._connect_strategy = connect_strategy
         self._client = client
-        self._user_session_token = user_session_token
 
-    def sql_credentials_provider(self, *args, **kwargs):
-        """The sql connector attempts to call the credentials provider w/o any args.
+        # TODO: set self._strategy during init and log the one that we picked
 
-        The SQL client's `ExternalAuthProvider` is not compatible w/ the SDK's implementation of
-        `CredentialsProvider`, so create a no-arg lambda that wraps the args defined by the real caller.
-        This way we can pass in a databricks `Config` object required by most of the SDK's `CredentialsProvider`
-        implementations from where `sql.connect` is called.
+    def auth_type(self) -> str:
+        """Returns the auth type currently in use.
+
+        The databricks-sdk client uses the configured auth_type to create
+        a user-agent string which is used for attribution. We should only
+        overwrite the auth_type if we are using the PositWorkbenchCredentialsStrategy
+        or PositConnectCredentialsStrategy (non-local), otherwise, we should return the auth_type
+        of the configured local_strategy instead to avoid breaking someone elses attribution.
+
+        NOTE: The databricks-sql client does not use auth_type to set the user-agent.
+        https://github.com/databricks/databricks-sql-python/blob/v3.3.0/src/databricks/sql/client.py#L214-L219
 
         See Also
         --------
-        * https://github.com/databricks/databricks-sql-python/issues/148#issuecomment-2271561365
-        """
-        return lambda: self.__call__(*args, **kwargs)
+        * https://github.com/databricks/databricks-sdk-py/blob/v0.29.0/databricks/sdk/config.py#L261-L269
 
-    def auth_type(self) -> str:
-        return _get_auth_type(self._local_strategy.auth_type())
+        Returns
+        -------
+            str
+        """
+        return self._auth_type
 
     def __call__(self, *args, **kwargs) -> CredentialsProvider:
-        # If the content is not running on Connect then fall back to local_strategy
-        if is_local():
+        if is_connect() and self._connect_strategy is not None:
+            self._auth_type = self._connect_strategy.auth_type()
+            return self._connect_strategy(*args, **kwargs)
+        else:
+            print()
+            # log and continue
```

Comment:
I'd like move these log messages into the init so that they aren't printed everytime the user requests credentials